cg/apps/deliverables_metrics_parser/parser/deliverables_parser.py

```diff
@@ -0,0 +1,122 @@
+"""Module to parse the metrics deliverables file."""
+import operator
+from enum import Enum
+from functools import reduce
+from pathlib import Path
+
+from cg.apps.deliverables_metrics_parser.models.pipeline_metrics_deliverables import (
+    MIPDNAMetricsDeliverables,
+)
+from cg.constants.constants import FileFormat
+from cg.constants.pipeline import Pipeline
+from cg.io.controller import ReadFile
+
+
+class MetricsParserConstants(Enum):
+    NAME: str = "name"
+    VALUE: str = "value"
+    ID: str = "id"
+    METRICS: str = "metrics"
+
+
+# we need some kind of standardisation across pipelines for these fields
+READ_INFORMATION_FIELDS = [
+    "reads_mapped",
+    "raw_total_sequences",
+    "percentage_mapped_reads",
+    "PCT_PF_READS_IMPROPER_PAIRS",
+]
+
+
+def generate_metrics_deliverables_model(
+    sample_id: str, sample_metrics: dict
+) -> MIPDNAMetricsDeliverables:
+    """Generate the metrics deliverable model for a sample."""
+    sample_metrics["id"] = sample_id
+    return MIPDNAMetricsDeliverables.model_validate(sample_metrics)
+
+
+def get_metrics_deliverables_file_path(pipeline: str, case_id: str) -> Path:
+    """Returns the metrics deliverables from the pipeline for a case."""
+    pipelines = {
+        Pipeline.MIPDNA: "/home/proj/production/rare-disease/cases/{case_id}/analysis/{case_id}_metrics_deliverables.yaml",
+        Pipeline.MIPRNA: "/home/proj/production/rare-disease/cases/{case_id}/analysis/{case_id}_metrics_deliverables.yaml",
+        Pipeline.BALSAMIC: "/home/proj/production/cancer/cases/{case_id}/analysis/qc/{case_id}_metrics_deliverables.yaml",
+        Pipeline.RNAFUSION: "/home/proj/production/rnafusion/cases/{case_id}/{case_id}_metrics_deliverables.yaml",
+        Pipeline.MUTANT: "/home/proj/production/mutant/cases/{case_id}/results/{case_id}_metrics_deliverables.yaml",
+    }
+
+    if pipeline in pipelines:
+        return Path(pipelines[pipeline].format(case_id=case_id))
+    else:
+        raise ValueError(f"Invalid pipeline name: {pipeline}")
+
+
+def read_metrics_deliverables(file_path: Path) -> list[dict]:
+    """Read the metrics deliverables file."""
+    return ReadFile.get_content_from_file(file_format=FileFormat.YAML, file_path=file_path)
+
+
+def get_metrics_from_content(content: list[dict[list[dict]]]) -> list[dict]:
```

Comment:
Could we create a model or data class for the list[dict[list[dict]]] object?

Reply:
These are the models that these files will be parsed into. At this stage files from multiple different pipelines can be used here. 

I can think of a general way in which a model can be used here.